protocols/gateway/handle.go

```diff
@@ -0,0 +1,91 @@
+package gateway
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	goHttp "net/http"
+
+	http "github.com/taubyte/http"
+	"github.com/taubyte/p2p/streams/client"
+	tunnel "github.com/taubyte/p2p/streams/tunnels/http"
+)
+
+func (g *Gateway) attach() {
+	g.http.LowLevel(&http.LowLevelDefinition{
+		PathPrefix: "/",
+		Handler: func(w goHttp.ResponseWriter, r *goHttp.Request) {
+			if err := g.handleHttp(w, r); err != nil {
+				w.Write([]byte(err.Error()))
+				w.WriteHeader(500)
+			}
+		},
+	})
+}
+
+func (g *Gateway) handleHttp(w goHttp.ResponseWriter, r *goHttp.Request) error {
+	resCh, err := g.substrateClient.ProxyHTTP(r.Host, r.URL.Path, r.Method)
```

Comment:
ProxyHandler(,,,timeout) {
...
New(....,Timeout(timeout)).Do()
...
}
---

for response := range resCh {
  
}